src/words/ast/INodeMovesPredicate.java

```diff
@@ -10,7 +15,22 @@ public INodeMovesPredicate(Object... children) {
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		ASTValue className = children.get(0).eval(environment);
```

Comment:
In the grammar right now, the "subject" of a basic action predicate can be either a class name *or* a particular object (using a reference list and identifier).  The first (0 index) child is always of type INodeSubject, but it looks like this implementation assumes it will only return a class name as a string.